### PR TITLE
Convert components to use custom props

### PR DESCRIPTION
**Pull request by cjwhitedev**
## Jira

[https://pegadigitalit.atlassian.net/browse/DS-36](https://pegadigitalit.atlassian.net/browse/DS-36)
[https://pegadigitalit.atlassian.net/browse/DS-115](https://pegadigitalit.atlassian.net/browse/DS-115)

## Summary

Update the intended web components to use css props instead of sass vars, mix-ins, and functions.

## Details

Mike is working on converting  bolt to css custom props, this PR is to convert all the components' sass files to use these props instead of the vars, mix-ins, and functions already in place. 

Updated components:

- bolt-accordion
- bolt-card
- bolt-carousel
- bolt-code-snippet
- bolt-copy-to-clipboard
- bolt-device-viewer
- bolt-icon
- bolt-image
- bolt-menu
- bolt-modal
- bolt-nav
- bolt-pagination
- bolt-popover
- bolt-progress-bar
- bolt-tabs
- bolt-toc
- bolt-tooltip
- bolt-trigger
- bolt-typeahead
- bolt-video

## How to test

- Open the doc site on both the [build](https://feature-bolt-spacing-css-custom-props-con.boltdesignsystem.com/) and on the live site to make sure both are identical. 
- Check that all the relevant components no longer use the outdated spacing methods (though i imagine the `@warn` will make this easier in time)

### CODE REVIEW

File: packages/components/bolt-pagination/src/pagination.scss [hunk 1]
```diff
@@ -53,6 +50,9 @@ $bolt-pagination-item-size: bolt-spacing(medium);
   -webkit-backface-visibility: hidden; // Removes the 1px jump with opacity transition
   transition: opacity $bolt-pagination-transition;
   white-space: nowrap;
+  @include bolt-font-family(body);
```
Comment (mikemai2awesome) [inline packages/components/bolt-pagination/src/pagination.scss:53]: Is linting putting all the includes after the props? They should be placed before.

File: packages/components/bolt-popover/src/popover.scss [hunk 2]
```diff
@@ -146,6 +142,10 @@ $bolt-popover-transition: $bolt-transition-timing
   background-color: bolt-theme(background);
   will-change: transform;
   transition: transform $bolt-popover-transition;
+  white-space: nowrap;
+  @include bolt-font-family(body);
```
Comment (mikemai2awesome) [inline packages/components/bolt-popover/src/popover.scss:146]: Same comment about the ordering of includes.

File: packages/components/bolt-tabs/src/tabs.scss [hunk 3]
```diff
@@ -201,61 +201,61 @@ bolt-tabs {
...
-          @include bolt-padding-right(0);
+          padding-right: 0;
+          padding-left: 0;
         }
         // first-of-type for no-JS
         &:first-child,
         &:first-of-type {
           .c-bolt-tabs__label-inner {
-            margin-left: -(bolt-spacing(#{$spacing-value-name}) / 2);
+            margin-left: (calc( var(--bolt-spacing-x-#{$size-name}) / -2 ));
```
Comment (mikemai2awesome) [inline packages/components/bolt-tabs/src/tabs.scss:233]: This should be, remove the overall ():
```
calc( var(--bolt-spacing-x-#{$size-name}) / -2 )
```

File: packages/components/bolt-tabs/src/tabs.scss [hunk 4]
```diff
@@ -201,61 +201,61 @@ bolt-tabs {
...
           }
         }
         // last-of-type for no-JS
         &:last-child,
         &:last-of-type {
           .c-bolt-tabs__label-inner {
-            margin-right: -(bolt-spacing(#{$spacing-value-name}) / 2);
+            margin-right: (calc( var(--bolt-spacing-x-#{$size-name}) / -2 ));
```
Comment (mikemai2awesome) [inline packages/components/bolt-tabs/src/tabs.scss:240]: Same here, remove the overall ().

File: packages/components/bolt-toc/src/_toc-item.scss [hunk 5]
```diff
@@ -5,16 +5,16 @@
...
-  @include bolt-font-size(small);
-  @include bolt-font-weight(regular);
   display: block;
   position: relative;
   transform: translate3d(0, 0, 0);
+  padding: calc(var(--bolt-spacing-y-small) / 2) var(--bolt-spacing-x-small);
   color: bolt-theme(text);
   text-decoration: none;
   transition: color $bolt-transition;
+  @include bolt-font-family(heading);
```
Comment (mikemai2awesome) [inline packages/components/bolt-toc/src/_toc-item.scss:15]: Same comment about includes ordering.

File: packages/components/bolt-toc/src/toc.scss [hunk 6]
```diff
@@ -33,25 +33,23 @@ bolt-toc {
...
-  @include bolt-padding(xsmall small);
-  @include bolt-font-family(heading);
-  @include bolt-font-size(large);
-  @include bolt-font-weight(bold);
   display: block;
+  margin: 0;
+  padding: var(--bolt-spacing-y-xsmall) var(--bolt-spacing-x-small);
   color: bolt-theme(background);
   background-color: bolt-theme(headline);
+  @include bolt-font-family(heading);
```
Comment (mikemai2awesome) [inline packages/components/bolt-toc/src/toc.scss:41]: Same comment about includes ordering.

File: packages/components/bolt-tooltip/src/tooltip.scss [hunk 7]
```diff
@@ -170,6 +167,9 @@ $bolt-tooltip-bubble-border-color: bolt-theme(background, bolt-opacity(20));
   will-change: transform;
   transition: transform $bolt-transition-timing
     cubic-bezier(0.175, 0.885, 0.32, 1);
+  @include bolt-font-family(body);
```
Comment (mikemai2awesome) [inline packages/components/bolt-tooltip/src/tooltip.scss:170]: Same comment about includes ordering.

File: packages/components/bolt-trigger/src/trigger.scss [hunk 8]
```diff
@@ -48,15 +48,15 @@ bolt-trigger {
...
-  @include bolt-font-family(body); // [1]
-  @include bolt-font-size(medium); // [2]
   display: inline; // Default to inline
   opacity: bolt-opacity(100);
+  margin: 0; // [1]
   color: inherit;
   text-align: inherit;
   text-decoration: initial;
   border: none; // [1]
+  @include bolt-font-family(body); // [1]
```
Comment (mikemai2awesome) [inline packages/components/bolt-trigger/src/trigger.scss:58]: Same comment about includes ordering.

File: packages/components/bolt-video/src/_videojs-enhancements.scss [hunk 9]
```diff
@@ -38,21 +38,21 @@ $bolt-video-js-button-shadow-level-hover: 'level-40';
...
+    bottom: var(--bolt-spacing-y-xsmall);
+    left: var(--bolt-spacing-x-xsmall);
     width: calc(1.2rem + 6%);
     height: calc(1rem + 6%);
+    margin: 0;
+    padding: 0;
     line-height: 1;
     border: none;
     border-radius: $bolt-video-js-button-border-radius;
+    @include bolt-video-js-button($opacity: bolt-opacity(100));
```
Comment (mikemai2awesome) [inline packages/components/bolt-video/src/_videojs-enhancements.scss:54]: Same comment about includes ordering.

File: packages/components/bolt-video/src/_videojs-enhancements.scss [hunk 10]
```diff
@@ -146,13 +146,13 @@ $bolt-video-js-button-shadow-level-hover: 'level-40';
...
     .vjs-modal-dialog-content {
-      @include bolt-padding(xsmall 2.5rem xsmall small);
-      @include bolt-font-size(medium, tight);
 
       display: flex;
       flex-wrap: wrap;
       align-items: center;
       overflow: auto;
+      padding: var(--bolt-spacing-y-xsmall) 2.5rem var(--bolt-spacing-y-xsmall) var(--bolt-spacing-x-small);
+      @include bolt-font-size(medium, tight);
```
Comment (mikemai2awesome) [inline packages/components/bolt-video/src/_videojs-enhancements.scss:155]: Same comment about includes ordering.

File: packages/components/bolt-video/src/_videojs-enhancements.scss [hunk 11]
```diff
@@ -164,10 +164,10 @@ $bolt-video-js-button-shadow-level-hover: 'level-40';
       }
 
       .vjs-social-description {
-        @include bolt-margin-bottom(0.5rem);
+        margin-bottom: 0.5rem;
+        white-space: nowrap;
         @include bolt-font-size(small, tight);
```
Comment (mikemai2awesome) [inline packages/components/bolt-video/src/_videojs-enhancements.scss:169]: Same comment about includes ordering.

File: packages/components/bolt-video/src/_videojs-enhancements.scss [hunk 12]
```diff
@@ -189,19 +189,19 @@ $bolt-video-js-button-shadow-level-hover: 'level-40';
       }
 
       form {
-        @include bolt-margin(auto);
-        @include bolt-font-family(body);
         width: 100%;
         max-width: 40rem;
+        margin: auto;
+        @include bolt-font-family(body);
```
Comment (mikemai2awesome) [inline packages/components/bolt-video/src/_videojs-enhancements.scss:195]: Same comment about includes ordering.

File: packages/components/bolt-video/src/_videojs-enhancements.scss [hunk 13]
```diff
@@ -290,18 +290,18 @@ $bolt-video-js-button-shadow-level-hover: 'level-40';
...
     width: auto;
     max-width: 100%;
+    margin: 0;
+    padding: var(--bolt-spacing-y-xsmall) var(--bolt-spacing-x-xsmall);
     color: bolt-color(black);
     text-shadow: 0 0 0.25rem bolt-color(white);
     border-radius: $bolt-border-radius;
     box-shadow: inset 0 0 2rem bolt-color(white), 0 0 0.25rem bolt-color(gray, light);
     background-color: rgba(bolt-color(white), 0.9);
+    @include bolt-font-family(body);
```
Comment (mikemai2awesome) [inline packages/components/bolt-video/src/_videojs-enhancements.scss:303]: Same comment about includes ordering.

**Comment by cjwhitedev** [comment]
I've been trying for a while and i cant seem to replicate the screengrab discrepancy locally. I was worried it might have been something I caused because it is spacing related, but since I cannot replicate it, I imagine it might be a failure on Travis' part.

File: packages/components/bolt-carousel/src/_carousel-slide.scss [hunk 14]
```diff
@@ -16,33 +16,35 @@ bolt-carousel-slide {
     flex-direction: column;
     flex-shrink: 1;
     box-sizing: content-box;
-    padding-right: $bolt-carousel-space-around;
     padding-right: var(--bolt-carousel-space-around);
```
[the remark this answers is not in the document]
Comment (danielamorse) [reply packages/components/bolt-carousel/src/_carousel-slide.scss:19]: Added ticket: https://pegadigitalit.atlassian.net/browse/DS-129

File: packages/components/bolt-navbar/src/navbar.scss [hunk 15]
```diff
@@ -125,60 +101,52 @@ bolt-navbar {
...
 
   &:only-child {
-    @include bolt-margin-right(auto);
+    margin-right: auto;
   }
 
-  @include bolt-if-browser-supports-display-contents {
-    @media screen and (max-width: $bolt-navbar-xsmall-bp) {
-      @include bolt-padding-top($bolt-navbar-vspacing-small);
-      grid-column: 2 / span 1;
```
[the remark this answers is not in the document]
Comment (mikemai2awesome) [reply packages/components/bolt-navbar/src/navbar.scss:165]: @danielamorse That's what I fixed last week. It's not needed. I've simplified it.

**Pull request merged**